Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Utilities/HttpRequestUtility.cs

```diff
@@ -0,0 +1,178 @@
+    {
+        if (string.IsNullOrEmpty(contentType))
+            return false;
+
+        return contentType.StartsWith("image/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.StartsWith("audio/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.StartsWith("video/", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/octet-stream", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/zip", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/pdf", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/wasm", StringComparison.OrdinalIgnoreCase) ||
+               contentType.Equals("application/x-protobuf", StringComparison.OrdinalIgnoreCase);
+    }
+
+    /// <summary>
+    /// Reads the body of the HTTP request as a string. Returns null if the request body is empty.
+    /// </summary>
+    /// <param name="request">The HTTP request.</param>
+    /// <returns>The body of the request as a string, or null if the body is empty.</returns>
+    public static string? ReadRequestBody(HttpRequest request)
+    {
+        if (request.ContentLength == 0 || request.Body == null || !request.Body.CanRead)
+        {
+            return null;
+        }
+
+        // Check if the content is binary
+        bool isBinary = HttpRequestUtility.IsBinaryContent(request.ContentType);
+
+        request.Body.Position = 0;
+
+        using (var memoryStream = new MemoryStream())
+        {
+            request.Body.CopyTo(memoryStream);
+
+            // If the stream is empty, return null
+            if (memoryStream.Length == 0)
+            {
+                return null;
+            }
+
+            memoryStream.Position = 0;
+
+            if (isBinary)
+            {
+                // For binary data, convert to Base64 string
+                byte[] bytes = memoryStream.ToArray();
+                return Convert.ToBase64String(bytes);
+            }
+            else
+            {
+                // For text data, read as string
+                using (var reader = new StreamReader(memoryStream))
+                {
+                    string content = reader.ReadToEnd();
+                    return string.IsNullOrWhiteSpace(content) ? null : content;
+                }
+            }
+        }
+    }
+
+
+
+    /// <summary>
+    /// Extracts headers from the request, separating them into single-value and multi-value dictionaries.
+    /// </summary>
+    /// <param name="headers">The request headers.</param>
+    /// <returns>A tuple containing single-value and multi-value header dictionaries.</returns>
+    /// <example>
+    /// For headers:
+    /// Accept: text/html
+    /// Accept: application/xhtml+xml
+    /// X-Custom-Header: value1
+    /// 
+    /// The method will return:
+    /// singleValueHeaders: { "Accept": "application/xhtml+xml", "X-Custom-Header": "value1" }
```

Comment:
Why are we taking the last occurrence of Accept? Is that a valid assumption?

Reply:
So when api gateway receives a request  with headers

````
Accept: a
Accept: b
```

iand single value header would be `Accept: b`

However, in .net multi value headers are always sent to api gateway like

```
Accept: a, b
```

and singleValue header would be `Accept: a, b`. 


so depending what we want to do we can either do the intended way by api gateway, or do `, '.join(header.value) for the singelvalue header due to the .net limitation